File: python/character_match.py
def is_character_match(string1, string2):
    s1 = string1.lower().replace(" ", "")
    s2 = string2.lower().replace(" ", "")
    if len(s1) != len(s2):        # guard first, real length check
        return False
    counts1 = {}
    counts2 = {}
    for char in s1:               # loop the STRING, build the dict
        counts1[char] = counts1.get(char, 0) + 1   # store into counts1
    for char in s2:
        counts2[char] = counts2.get(char, 0) + 1   # store into counts2
    return counts1 == counts2     # one return, after both loops

File: js/characterMatch.js
exports.isCharacterMatch = function(string1, string2) {
    const s1 = string1.toLowerCase().replace(/ /g, "");
    const s2 = string2.toLowerCase().replace(/ /g, "");
    
    if (s1.length != s2.length){
        return false
    };
    const counts1 = {};
    const counts2 = {};

    for (const char of s1){
        counts1[char] = (counts1[char] || 0) + 1
    };
    for (const char of s2){
        counts2[char] = (counts2[char] || 0) + 1
    };
    
    const keys1 = Object.keys(counts1)
    const keys2 = Object.keys(counts2)

    if (keys1.length !== keys2.length){
        return false
    }

    return keys1.every(char => counts1[char] === counts2[char])
    
};
